pkg/controller/mutators/core/reconciler.go

```diff
@@ -271,74 +243,111 @@ func (r *Reconciler) getOrDefault(ctx context.Context, namespacedName apiTypes.N
 		// Treat objects with a DeletionTimestamp as if they are deleted.
 		deleted := !obj.GetDeletionTimestamp().IsZero()
 		return obj, deleted, nil
-	case errors.IsNotFound(err):
+	case apierrors.IsNotFound(err):
 		obj = r.newMutationObj()
 		obj.SetName(namespacedName.Name)
 		obj.SetNamespace(namespacedName.Namespace)
-		obj.GetObjectKind().SetGroupVersionKind(r.gvk())
+		obj.GetObjectKind().SetGroupVersionKind(r.gvk)
 		return obj, true, nil
 	default:
 		return nil, false, err
 	}
 }
 
 func (r *Reconciler) getTracker() readiness.Expectations {
-	return r.tracker.For(r.gvk())
+	return r.tracker.For(r.gvk)
 }
 
 // reconcileDeleted removes the Mutator from the controller and deletes the corresponding PodStatus.
-func (r *Reconciler) reconcileDeleted(ctx context.Context, obj client.Object) (reconcile.Result, error) {
-	mID := types.MakeID(obj)
-	r.getTracker().CancelExpect(obj)
+func (r *Reconciler) reconcileDeleted(ctx context.Context, mID types.ID) error {
 	r.cache.Remove(mID)
 
 	if err := r.system.Remove(mID); err != nil {
 		r.log.Error(err, "Remove failed", "resource",
-			apiTypes.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()})
-		return reconcile.Result{}, err
+			apiTypes.NamespacedName{Name: mID.Name, Namespace: mID.Namespace})
+		return err
 	}
 
-	sName, err := statusv1beta1.KeyForMutatorID(util.GetPodName(), mID)
+	pod, err := r.getPod(ctx)
 	if err != nil {
-		return reconcile.Result{}, err
+		return err
+	}
+
+	sName, err := statusv1beta1.KeyForMutatorID(pod.Name, mID)
+	if err != nil {
+		return err
 	}
 
 	status := &statusv1beta1.MutatorPodStatus{}
 	status.SetName(sName)
-	status.SetNamespace(util.GetNamespace())
-	if err = r.Delete(ctx, status); !errors.IsNotFound(err) {
-		return reconcile.Result{}, err
+	status.SetNamespace(pod.Namespace)
+	if err = r.Delete(ctx, status); err != nil && !apierrors.IsNotFound(err) {
+		return err
 	}
 
-	return reconcile.Result{}, nil
+	return nil
+}
+
+// queueConflicts queues updates for Mutators in ids except for ignore.
```

Comment:
nit: "except for ignore." -- is that part of the comment still valid?

Reply:
Good catch.